tqecjs/src/workspace/plaquettes/PlaquetteClass.js

```diff
@@ -0,0 +1,370 @@
+				lowestYQubit = sortedQubits[q];
+			}
+		}
+		// Sort points by polar angle with respect to lowestYQubit
+		let sortedPoints = sortedQubits.toSorted((a, b) => {
+			// Calculate the polar angle
+			const polarAngleA = Math.atan2(
+				a.globalY - lowestYQubit.globalY,
+				a.globalX - lowestYQubit.globalX
+			);
+			const polarAngleB = Math.atan2(
+				b.globalY - lowestYQubit.globalY,
+				b.globalX - lowestYQubit.globalX
+			);
+			a.pAngle = polarAngleA;
+			b.pAngle = polarAngleB;
+
+			// Get the distance from the lowestYQubit
+			const distA = Math.sqrt(
+				Math.pow(a.globalX - lowestYQubit.globalX, 2) +
+					Math.pow(a.globalY - lowestYQubit.globalY, 2)
+			);
+			const distB = Math.sqrt(
+				Math.pow(b.globalX - lowestYQubit.globalX, 2) +
+					Math.pow(b.globalY - lowestYQubit.globalY, 2)
+			);
+			a.dist = distA;
+			b.dist = distB;
+			return polarAngleB - polarAngleA;
+		});
+
+		sortedPoints = sortedPoints.filter((qubit) => {
+			// If the qubit is the lowestYQubit, then keep it
+			if (qubit === lowestYQubit) return qubit;
+			// Get the qubits that have the same polar angle
+			const samepQubits = sortedPoints.filter((q) => q.pAngle === qubit.pAngle);
+			// Get the maximum distance qubit
+			const maxDistQubit = samepQubits.sort((a, b) => {
+				return b.dist - a.dist;
+			})[0];
+			// Return the qubit with the maximum distance
+			return qubit === maxDistQubit;
+		});
+		// Create the stack of qubits which contains the convex hull
+		for (const qubit of sortedPoints) {
+			// Remove qubits from the stack if the angle formed by points next-to-top, top, and qubit is not counterclockwise
+			while (
+				qubitStack.length > 1 &&
+				this._ccw(
+					qubitStack.at(-2), // second from top
+					qubitStack.at(-1), // top of stack
+					qubit
+				) <= 0
+			) {
+				qubitStack.pop();
+			}
+			// push the qubit onto the stack
+			qubitStack.push(qubit);
+		}
+		// Map the qubit stack to the global coordinates
+		const qubitPos = qubitStack.map((qubit) => ({
+			x: qubit.globalX,
+			y: qubit.globalY,
+		}));
+
+		this.beginFill(this.color);
+		// Fill the convex hull
+		this.drawPolygon(qubitPos);
+		this.cursor = 'pointer';
+		this.endFill();
+		// Assign the qubit stack to the plaquette
+		this.qubitStack = qubitStack;
+	};
+
+	_ccw = (p, q, r) => {
+		// Check if the points are counterclockwise
+		const v1 = { x: q.globalX - p.globalX, y: q.globalY - p.globalY };
+		const v2 = { x: r.globalX - q.globalX, y: r.globalY - q.globalY };
+		// Take the cross product
+		const val = -(v1.x * v2.y - v1.y * v2.x); // If val > 0, then counterclockwise, else clockwise
+		if (val === 0) return 0; // Collinear
+		return val; // Clockwise  < 0 or counterclockwise >0
+	};
+
+	createPlaquette = () => {
+		// The graphic should connect the points from the qubits
+		const nQubits = this.qubits.length;
+		if (nQubits < 3) {
+			console.log('Plaquette must have at least 3 qubits');
+			this.plaquetteMade = false;
```

Comment:
Would it make sense to throw an exception here?

Reply:
I think it would make more sense to use the notifier and tell the user to select another qubit

Reply:
Done, may need some testing and formatting